docs/source/examples/checks/methodology/single_feature_contribution_train_test.ipynb

```diff
@@ -5,123 +5,98 @@
    "id": "21c5d5bc",
```

Comment:
<p>The check estimates for every feature its ability to predict the label by itself. This check can help find:</p><ul><li>A potential leakage (between the label and a feature) in both datasets - e.g. due to incorrect sampling during data collection. This is a critical problem, that will likely stay hidden without this check (as it won't pop up when comparing model performance on train and test).</li><li>A strong drift between the the feature-label relation in both datasets, possibly originating from a leakage in one of the datasets - e.g. a leakage that exists in the training data, but not necessarily in a "fresh" dataset, that may have been built differently.</li></ul><p> </p><p>The check is based on calculating the predictive power score (PPS) of each feature. For more details you can read here [how the PPS is calculated[#how-is-the-predictive-power-score-(pps)-calculated?]. </p><p>(the above should be an internal link to next section... which I guess should be in a different cell)</p><p> </p><h3>What is a Problematic Result?</h3><p>1. <strong>Features with a high predictive score</strong> - can indicate that there is a leakage between the label and the feature, meaning that the feature holds information that is somewhat based on the label to begin with. For example: a bank uses their loans database to create a model of whether a customer will be able to return a loan. One of the features they extract is "number of late payments". It is clear this feature will have a very strong prediction power on the customer's ability to return his loan, but this feature is based on data the bank knows only <em>after</em> the loan is given, so it won't be available during the time of the prediction, and is a type of leakage.</p><p>2. <strong>A high difference between the PPS scores of a certain feature in the train and in the test datasets</strong> - this is an indication for a drift between the relation of the feature and the label and a possible leakage in one of the datasets. For example: an e-commerce shop added a new department (e.g. sports equipment) and it tries to predict if a user will buy an item from that department this week. If the training data is based on the first two weeks of the departments' operation, it is likely that when the feature "time_department_last_viewed" is not None, this will be a strong indicator for the label being "True", while if the test data is further on in the future, having the "time_department_last_viewed" be not None is a much weaker indicator for user buying something from there. Indicating that the data sampling and model training should be adjusted to fix this relative leakage in the training data.</p><p> </p><h3>How is the Predictive Power Score (PPS) Calculated?</h3><p>The features' predictive score results in a numeric score between 0 (feature has no predictive power) and 1 (feature can fully predict the label alone).\</p><p>The process of calculating the PPS is the following:</p><p>... all of the points..</p><p>*note to change the "5000" so it's clear that it's the number of samples taht can be updated...</p><p> </p><p> </p><p>...</p><p>For further information about PPS you can visit the <a href="https://github.com/8080labs/ppscore" target="_blank">ppscore github</a> or the following blog post: <a href="https://towardsdatascience.com/rip-correlation-introducing-the-predictive-power-score-3d90808b9598" target="_blank">RIP correlation. Introducing the Predictive Power Score</a></p><p> </p><p>**(and makeing sure that these are external links that open in new tab?)</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/deepchecks/deepchecks/pull/642/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>